src/Sentry/SentryLog.cs

```diff
@@ -0,0 +1,281 @@
+    /// Returns <see langword="true"/> if the <see cref="SentryLog"/> contains an attribute with the specified key of type <see cref="long"/>.
+    /// Otherwise <see langword="false"/>.
+    /// </remarks>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public bool TryGetAttribute(string key, out long value)
+    {
+        if (_attributes.TryGetValue(key, out var attribute) && attribute.Type == "integer")
+        {
+            value = (long)attribute.Value;
+            return true;
+        }
+
+        value = 0L;
+        return false;
+    }
+
+    /// <summary>
+    /// Gets the attribute value associated with the specified key when of type <see cref="double"/>.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    /// <remarks>
+    /// Returns <see langword="true"/> if the <see cref="SentryLog"/> contains an attribute with the specified key of type <see cref="double"/>.
+    /// Otherwise <see langword="false"/>.
+    /// </remarks>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public bool TryGetAttribute(string key, out double value)
+    {
+        if (_attributes.TryGetValue(key, out var attribute) && attribute.Type == "double")
+        {
+            value = (double)attribute.Value;
+            return true;
+        }
+
+        value = 0.0;
+        return false;
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="string"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, string value)
+    {
+        _attributes[key] = new SentryAttribute(value, "string");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="bool"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, bool value)
+    {
+        _attributes[key] = new SentryAttribute(value, "boolean");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="long"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, long value)
+    {
+        _attributes[key] = new SentryAttribute(value, "integer");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="double"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, double value)
+    {
+        _attributes[key] = new SentryAttribute(value, "double");
+    }
+
+    internal void SetAttributes(SentryOptions options)
```

Comment:
I mentioned at a callsite that I couldn't tell what was going on from the method name and the fact it took options. It's internal so not a big deal but perhaps `SetDefaultAttributes` or `SetBaseAttributes`

Reply:
Cood call! I like `SetDefaultAttributes` as per specification/documentation.